src/main/java/com/sinthoras/hydroenergy/client/renderer/HEProgram.java

```diff
@@ -59,7 +59,7 @@
 
 
     public static void init() {
-        if(GLContext.getCapabilities().OpenGL32 && !HEConfig.useLimitedRendering) {
+        if((GLContext.getCapabilities().OpenGL32 && !HEConfig.useLimitedRendering) || HEConfig.forceOpenGL) {
```

Comment:
Where did you learn about it? Did you just try it?

Reply:
Apparently MacOS lies about the OpenGL version: https://stackoverflow.com/questions/19658745/why-is-my-opengl-version-always-2-1-on-mac-os-x - so this was a test to see if it would "run" - with this the spamming OpenGL errors go away.

Reply:
That would mean that `GLContext.getCapabilities().OpenGL32` produces the error? It should be skipped with the override flag.

Reply:
It doesn't produce the spamming error, but it returns `false` on macOS when it should return true. I can rework the IF statement - suggestions? Maybe move the force next to the getcapabilities?